app/assets/javascripts/components/revisions/revisions_handler.jsx

```diff
@@ -115,6 +124,15 @@ const RevisionHandler = createReactClass({
         />
         {!loaded && <Loading/>}
         {loaded && showMoreButton}
+        {loaded && metaDataLoading && (
+          <div id="progress" >
```

Comment:
You should extract this as a new component that takes the loading message as its prop, so that we can easily re-use this in other places too.